[Bug]: `yarn start` for `@fluentui/react-components` fails to start

### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
System:
    OS: Windows 10 10.0.22000
    CPU: (8) x64 Intel(R) Core(TM) i7-1065G7 CPU @ 1.30GHz
    Memory: 12.90 GB / 31.59 GB
Browsers:
    Edge: Spartan (44.22000.120.0), Chromium (104.0.1293.63), ChromiumDev (106.0.1349.1)
    Internet Explorer: 11.0.22000.120
```


### Are you reporting Accessibility issue?

no

### Reproduction

0. clone fluent repo 1. `yarn start` 2. select `@fluentui/react-components`

NB: I'm at commit: 3c89bbf0dd

### Bug Description

## Actual Behavior

Running `yarn start` > `@fluentui/react-components` results in an error.

It looks like the issue is with this path: `Error: Cannot find module '..src/preset/preview'`. I would expect the actual path to be `../src/preset/preview` but I'm not 100% certain this is the case. I was only able to find this path in a test so I'm not sure where it's actually set.

Full error log in the logs field for the bug report.


## Expected Behavior

Running `yarn start` > `@fluentui/react-components` should launch Storybook.


### Logs

```shell
[seanmonahan] dev [fix/color-token-for-appearance-backgrounds]> yarn start
yarn run v1.23.34
$ node scripts/start.js
√ Which project to start (select or type partial name)? » @fluentui/react-components
$ yarn storybook
$ start-storybook --port 3000 --no-manager-cache
info @storybook/react v6.5.5
info
info => Loading presets
info => Serving static files from ./.\public at /
info => Loading custom manager config
ERR! Error: Cannot find module '..src/preset/preview'
ERR! Require stack:
ERR! - C:\Users\seanmonahan\Documents\dev\fluentui\packages\react-components\react-storybook-addon\dist\preset.js
ERR! - C:\Users\seanmonahan\Documents\dev\fluentui\node_modules\@storybook\core-common\dist\cjs\presets.js
ERR! - C:\Users\seanmonahan\Documents\dev\fluentui\node_modules\@storybook\core-common\dist\cjs\index.js
ERR! - C:\Users\seanmonahan\Documents\dev\fluentui\node_modules\@storybook\core-server\dist\cjs\index.js
ERR! - C:\Users\seanmonahan\Documents\dev\fluentui\node_modules\@storybook\core\dist\cjs\server.js
ERR! - C:\Users\seanmonahan\Documents\dev\fluentui\node_modules\@storybook\core\server.js
ERR! - C:\Users\seanmonahan\Documents\dev\fluentui\node_modules\@storybook\react\dist\cjs\server\index.js
ERR! - C:\Users\seanmonahan\Documents\dev\fluentui\node_modules\@storybook\react\bin\index.js
ERR!     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:902:15)
ERR!     at Function.Module._resolveFilename (C:\Users\seanmonahan\Documents\dev\fluentui\node_modules\tsconfig-paths\lib\register.js:75:40)
ERR!     at Function.resolve (internal/modules/cjs/helpers.js:99:19)
ERR!     at Object.config (C:\Users\seanmonahan\Documents\dev\fluentui\packages\react-components\react-storybook-addon\dist\preset.js:7:27)
ERR!     at C:\Users\seanmonahan\Documents\dev\fluentui\node_modules\@storybook\core-common\dist\cjs\presets.js:294:28
ERR!     at processTicksAndRejections (internal/process/task_queues.js:95:5)
ERR!     at _default (C:\Users\seanmonahan\Documents\dev\fluentui\node_modules\@storybook\builder-webpack5\dist\cjs\preview\iframe-webpack.config.js:70:22)
ERR!     at starterGeneratorFn (C:\Users\seanmonahan\Documents\dev\fluentui\node_modules\@storybook\builder-webpack5\dist\cjs\index.js:116:16)
ERR!     at Object.start (C:\Users\seanmonahan\Documents\dev\fluentui\node_modules\@storybook\builder-webpack5\dist\cjs\index.js:272:14)
ERR!     at async Promise.all (index 0)
ERR!     at storybookDevServer (C:\Users\seanmonahan\Documents\dev\fluentui\node_modules\@storybook\core-server\dist\cjs\dev-server.js:207:28)
ERR!     at buildDevStandalone (C:\Users\seanmonahan\Documents\dev\fluentui\node_modules\@storybook\core-server\dist\cjs\build-dev.js:120:31)
ERR!     at buildDev (C:\Users\seanmonahan\Documents\dev\fluentui\node_modules\@storybook\core-server\dist\cjs\build-dev.js:174:5)
ERR!  Error: Cannot find module '..src/preset/preview'
ERR! Require stack:
ERR! - C:\Users\seanmonahan\Documents\dev\fluentui\packages\react-components\react-storybook-addon\dist\preset.js
ERR! - C:\Users\seanmonahan\Documents\dev\fluentui\node_modules\@storybook\core-common\dist\cjs\presets.js
ERR! - C:\Users\seanmonahan\Documents\dev\fluentui\node_modules\@storybook\core-common\dist\cjs\index.js
ERR! - C:\Users\seanmonahan\Documents\dev\fluentui\node_modules\@storybook\core-server\dist\cjs\index.js
ERR! - C:\Users\seanmonahan\Documents\dev\fluentui\node_modules\@storybook\core\dist\cjs\server.js
ERR! - C:\Users\seanmonahan\Documents\dev\fluentui\node_modules\@storybook\core\server.js
ERR! - C:\Users\seanmonahan\Documents\dev\fluentui\node_modules\@storybook\react\dist\cjs\server\index.js
ERR! - C:\Users\seanmonahan\Documents\dev\fluentui\node_modules\@storybook\react\bin\index.js
ERR!     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:902:15)
ERR!     at Function.Module._resolveFilename (C:\Users\seanmonahan\Documents\dev\fluentui\node_modules\tsconfig-paths\lib\register.js:75:40)
ERR!     at Function.resolve (internal/modules/cjs/helpers.js:99:19)
ERR!     at Object.config (C:\Users\seanmonahan\Documents\dev\fluentui\packages\react-components\react-storybook-addon\dist\preset.js:7:27)
ERR!     at C:\Users\seanmonahan\Documents\dev\fluentui\node_modules\@storybook\core-common\dist\cjs\presets.js:294:28
ERR!     at processTicksAndRejections (internal/process/task_queues.js:95:5)
ERR!     at _default (C:\Users\seanmonahan\Documents\dev\fluentui\node_modules\@storybook\builder-webpack5\dist\cjs\preview\iframe-webpack.config.js:70:22)
ERR!     at starterGeneratorFn (C:\Users\seanmonahan\Documents\dev\fluentui\node_modules\@storybook\builder-webpack5\dist\cjs\index.js:116:16)
ERR!     at Object.start (C:\Users\seanmonahan\Documents\dev\fluentui\node_modules\@storybook\builder-webpack5\dist\cjs\index.js:272:14)
ERR!     at async Promise.all (index 0)
ERR!     at storybookDevServer (C:\Users\seanmonahan\Documents\dev\fluentui\node_modules\@storybook\core-server\dist\cjs\dev-server.js:207:28)
ERR!     at buildDevStandalone (C:\Users\seanmonahan\Documents\dev\fluentui\node_modules\@storybook\core-server\dist\cjs\build-dev.js:120:31)
ERR!     at buildDev (C:\Users\seanmonahan\Documents\dev\fluentui\node_modules\@storybook\core-server\dist\cjs\build-dev.js:174:5) {
ERR!   code: 'MODULE_NOT_FOUND',
ERR!   requireStack: [
ERR!     'C:\\Users\\seanmonahan\\Documents\\dev\\fluentui\\packages\\react-components\\react-storybook-addon\\dist\\preset.js',
ERR!     'C:\\Users\\seanmonahan\\Documents\\dev\\fluentui\\node_modules\\@storybook\\core-common\\dist\\cjs\\presets.js',
ERR!     'C:\\Users\\seanmonahan\\Documents\\dev\\fluentui\\node_modules\\@storybook\\core-common\\dist\\cjs\\index.js',
ERR!     'C:\\Users\\seanmonahan\\Documents\\dev\\fluentui\\node_modules\\@storybook\\core-server\\dist\\cjs\\index.js',
ERR!     'C:\\Users\\seanmonahan\\Documents\\dev\\fluentui\\node_modules\\@storybook\\core\\dist\\cjs\\server.js',
ERR!     'C:\\Users\\seanmonahan\\Documents\\dev\\fluentui\\node_modules\\@storybook\\core\\server.js',
ERR!     'C:\\Users\\seanmonahan\\Documents\\dev\\fluentui\\node_modules\\@storybook\\react\\dist\\cjs\\server\\index.js',
ERR!     'C:\\Users\\seanmonahan\\Documents\\dev\\fluentui\\node_modules\\@storybook\\react\\bin\\index.js'
ERR!   ]
ERR! }
```
```


### Requested priority

Blocking

### Products/sites affected

_No response_

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

--- a/scripts/storybook/utils.js
+++ b/scripts/storybook/utils.js
@@ -36,7 +36,7 @@ function loadWorkspaceAddon(addonName, options = {}) {
     const tsConfigPath = path.join(rootPath, 'tsconfig.lib.json');
     const packageJsonPath = path.join(rootPath, 'package.json');
     const sourceRootPath = path.join(workspaceRoot, addonMetadata.sourceRoot);
-    console.log('sourceRootPath', sourceRootPath);
+
     /**
      * @type {Record<string,any>}
      */
@@ -47,7 +47,7 @@ function loadWorkspaceAddon(addonName, options = {}) {
     /**
      * we use always POSIX path in js modules, thus this needs to be explicit for all OS, to properly replace content
      */
-    console.log('distPath', distPath);
+
     const relativePathToSource = path.relative(distPath, sourceRootPath).split(path.sep).join(path.posix.sep);
     const presetSourcePath = path.join(rootPath, 'preset.js');
     const presetMemorySourcePath = path.join(distPath, 'preset.js');
@@ -69,13 +69,11 @@ function loadWorkspaceAddon(addonName, options = {}) {
     );
   }
 
-  console.log('relativePathToSource', relativePathToSource);
   const presetContent = fs.readFileSync(presetSourcePath, 'utf-8');
 
   const regex = new RegExp(`\\./${path.normalize(packageDistPath)}`, 'g');
   let modifiedPresetContent = presetContent.replace(regex, relativePathToSource);
 
-  console.log('modifiedPresetContent', modifiedPresetContent);
   modifiedPresetContent = stripIndents`
     const { workspaceRoot } = require('nx/src/utils/app-root');
     const { registerTsProject } = require('nx/src/utils/register');

--- a/scripts/storybook/utils.js
+++ b/scripts/storybook/utils.js
@@ -36,6 +36,7 @@ function loadWorkspaceAddon(addonName, options = {}) {
     const tsConfigPath = path.join(rootPath, 'tsconfig.lib.json');
     const packageJsonPath = path.join(rootPath, 'package.json');
     const sourceRootPath = path.join(workspaceRoot, addonMetadata.sourceRoot);
+    console.log('sourceRootPath', sourceRootPath);
     /**
      * @type {Record<string,any>}
      */
@@ -46,7 +47,8 @@ function loadWorkspaceAddon(addonName, options = {}) {
     /**
      * we use always POSIX path in js modules, thus this needs to be explicit for all OS, to properly replace content
      */
-    const relativePathToSource = path.posix.relative(distPath, sourceRootPath);
+    console.log('distPath', distPath);
+    const relativePathToSource = path.relative(distPath, sourceRootPath).split(path.sep).join(path.posix.sep);
     const presetSourcePath = path.join(rootPath, 'preset.js');
     const presetMemorySourcePath = path.join(distPath, 'preset.js');
 
@@ -67,11 +69,13 @@ function loadWorkspaceAddon(addonName, options = {}) {
     );
   }
 
+  console.log('relativePathToSource', relativePathToSource);
   const presetContent = fs.readFileSync(presetSourcePath, 'utf-8');
 
   const regex = new RegExp(`\\./${path.normalize(packageDistPath)}`, 'g');
   let modifiedPresetContent = presetContent.replace(regex, relativePathToSource);
 
+  console.log('modifiedPresetContent', modifiedPresetContent);
   modifiedPresetContent = stripIndents`
     const { workspaceRoot } = require('nx/src/utils/app-root');
     const { registerTsProject } = require('nx/src/utils/register');

--- a/scripts/storybook/utils.js
+++ b/scripts/storybook/utils.js
@@ -36,7 +36,6 @@ function loadWorkspaceAddon(addonName, options = {}) {
     const tsConfigPath = path.join(rootPath, 'tsconfig.lib.json');
     const packageJsonPath = path.join(rootPath, 'package.json');
     const sourceRootPath = path.join(workspaceRoot, addonMetadata.sourceRoot);
-
     /**
      * @type {Record<string,any>}
      */
@@ -47,7 +46,6 @@ function loadWorkspaceAddon(addonName, options = {}) {
     /**
      * we use always POSIX path in js modules, thus this needs to be explicit for all OS, to properly replace content
      */
-
     const relativePathToSource = path.relative(distPath, sourceRootPath).split(path.sep).join(path.posix.sep);
     const presetSourcePath = path.join(rootPath, 'preset.js');
     const presetMemorySourcePath = path.join(distPath, 'preset.js');

